File: algoritmo_gulosos/estacao_radio.dart
void main(List<String> args) {
  Set<String> estados_abrange = Set<String>.from(["Acre", "Alagoas", "Bahia", "Goiás", "Paraná", "Tocantins", "São Paulo", "Rio de Janeiro"]);

  var estacoes = {};
  estacoes["kun"] = Set<String>.from(["Goiás", "Paraná", "Tocantins"]);
  estacoes["kdois"] = Set<String>.from(["Alagoas", "Goiás", "Acre"]);
  estacoes["ktres"] = Set<String>.from(["Bahia", "Paraná", "São Paulo"]);
  estacoes["kquatro"] = Set<String>.from(["Paraná", "Tocantins"]);
  estacoes["kcinco"] = Set<String>.from(["São Paulo", "Rio de Janeiro"]);

  Set estacoes_finais = Set<String>();

  while (estados_abrange.isNotEmpty) {
    Set melhor_estacao = Set<String>();
    Set estados_cobertos = Set<String>();

    for (var item in estacoes.values) {
      Set cobertos = estados_abrange.intersection(item);

      if (cobertos.length > estados_cobertos.length) {
        melhor_estacao = item;
        estados_cobertos = cobertos;
      }

      estados_abrange.removeAll(estados_cobertos);
      estacoes_finais.addAll(melhor_estacao);
    }
  }

  print(estacoes_finais);
}

File: algoritmo_gulosos/estacao_radio.py
estados_abranger = set(["Acre","Alagoas","Bahia","Goiás", "Paraná", "Tocantins", "São Paulo", "Rio de Janeiro"])

estacoes = {}
estacoes["kun"] = set(["Goiás", "Paraná", "Tocantins"])
estacoes["kdois"] = set(["Alagoas", "Goiás", "Acre"])
estacoes["ktres"] = set(["Bahia", "Paraná", "São Paulo"])
estacoes["kquatro"] = set(["Paraná", "Tocantins"])
estacoes["kcinco"] = set(["São Paulo", "Rio de Janeiro"])

estacoes_finais = set()

while estados_abranger:
    melhor_estacao = None
    estados_cobertos = set()

    for estacao, estados in estacoes.items():
        cobertos = estados_abranger & estados
        if len(cobertos) > len(estados_cobertos):
            melhor_estacao = estacao
            estados_cobertos = cobertos

    estados_abranger -= estados_cobertos
    estacoes_finais.add(melhor_estacao)


print(estacoes_finais)
